os/arch/arm/src/amebad/amebad_watchdog_lowerhalf.c

```diff
@@ -0,0 +1,482 @@
+		watchdog_refresh();
+
+		/* Update start time and reset on_pause_duration */
+		priv->start_tick = SYSTIMER_TickGet();
+		priv->on_pause_duration = 0;
+	}
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: amebad_wdg_getstatus
+ *
+ * Description:
+ *   Get the current watchdog timer status
+ *
+ * Input Parameters:
+ *   lower  - A pointer the publicly visible representation of the "lower-half"
+ *            driver state structure.
+ *   status - The location to return the watchdog status information.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int amebad_wdg_getstatus(FAR struct watchdog_lowerhalf_s *lower, FAR struct watchdog_status_s *status)
+{
+	struct amebad_wdg_lowerhalf_s *priv = (struct amebad_wdg_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+	if (!priv) {
+		return -ENODEV;
+	}
+
+	/* Reture watchdog flag */
+	status->flags = WDFLAGS_RESET;
+
+	/* Return ACTIVE if watchdog timer is started */
+	if (priv->wdt_status == WDT_START) {
+		status->flags |= WDFLAGS_ACTIVE;
+	}
+
+	/* Return CAPTURE if interrupt mode is enabled */
+	if (priv->int_mode) {
+		status->flags |= WDFLAGS_CAPTURE;
+	}
+
+	/* Return WAIT if watchdog timer is on pause */
+	if (priv->wdt_status == ON_PAUSE) {
+		status->flags |= WDFLAGS_WAIT;
+	}
+
+	/* Reture timeout in ms */
+	status->timeout = priv->timeout_ms;
+
+	/* Reture the timeleft before watchdog timeout in ms */
+	if (priv->wdt_status == ON_PAUSE) {
+		status->timeleft = priv->timeout_left;
+	} else {
+		uint32_t time_elapse;
+		time_elapse = ((SYSTIMER_TickGet() - priv->start_tick - priv->on_pause_duration) * 31) / 1000;	// Get time elapse in ms
+		status->timeleft = status->timeout - time_elapse;
+	}
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: amebad_wdg_settimeout
+ *
+ * Description:
+ *   Set a new timeout value (and reset the watchdog timer)
+ *
+ * Input Parameters:
+ *   lower   - A pointer the publicly visible representation of the
+ *             "lower-half" driver state structure.
+ *   timeout - The new timeout value in milliseconds.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int amebad_wdg_settimeout(FAR struct watchdog_lowerhalf_s *lower, uint32_t timeout)
+{
+	struct amebad_wdg_lowerhalf_s *priv = (struct amebad_wdg_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+	if (!priv) {
+		return -ENODEV;
+	}
+
+	/*  Reset the watchdog timer with new timeout value */
+	priv->timeout_ms = timeout;
+	watchdog_init(priv->timeout_ms);
+	if (priv->int_mode) {
+		watchdog_irq_init((void *)priv->handler, (uint32_t)priv);
+	}
+
+	/* Re-start watchdog only if the status is WDT_START */
+	if (priv->wdt_status == WDT_START) {
```

Comment:
Is this ok to re-set above without stopping in this case?

Reply:
Yes, every time using **watchdog_init**, the watchdog register will be reset, including the watchdog enable bit that is used by watchdog_stop. 
Also, if '**init**' without a '**start**' (no matter if watchdog is already started before), the watchdog is not working.  

In this case, if watchdog is started, '**init**' + '**start**'; If watchdog is on pause, '**init**' only, and '**start**' is only done by WDIOC_RESUME; If watchdog is stopped, '**init**' only, and '**start**' is only done by **amebad_wdg_start**.